File: problems/2/scala/src/com/github/rogeralmeida/eulersjourney/problem2/EvenFibonacciSum.scala
package com.github.rogeralmeida.eulersjourney.problem2

object EvenFibonacciSum {

  def main(args: Array[String]) {
    lazy val fib: Stream[Int] = 0 #:: 1 #:: fib.zip(fib.tail).map { case (a, b) => a + b}
    var total = 0
    fib.filter(number => number % 2 == 0).toStream.takeWhile(number => number < 4000000).foreach(number => total += number)
    printf("The some of even numbers in fibonacci sequence bellow 4000 is %d", total)
    
  }

}

File: problems/2/javascript/evenFibonacciSum.js
function fib(n) {
  return function(n,a,b) {
    return n>0 ? arguments.callee(n-1,b,a+b) : a;
  }(n,0,1);
}
number = 0;
total = 0;
i = 0;
while (number < 4000000){
  number = fib(i++);
  if(number%2 == 0){
    total += number;
  }
}

console.log("The sum of even numbers in fibonacci sequence bellow 4kk is "+total);
